06_gpu/stable_diffusion_cli.py

```diff
@@ -0,0 +1,118 @@
+# ---
+# output-directory: "/tmp/stable-diffusion"
+# ---
+# # Stable Diffusion CLI
+#
+# This tutorial shows how you can create a CLI tool that runs GPU-intensive
+# work remotely but feels like you are running locally. We will be building
+# a tool that generates an image based on a prompt against Stable Diffusion
+# using the HuggingFace Hub and the `diffusers` library.
+
+# ## Basic setup
+import modal
+from pathlib import Path
+
+# All Modal programs need a [`Stub`](/docs/reference/modal.Stub) — an object that acts as a recipe for
+# the application. Let's give it a friendly name.
+
+stub = modal.Stub("stable-diffusion-cli")
+
+# We will be using `typer` to create our CLI interface.
+
+import typer
+
+app = typer.Typer()
+
+# ## Model dependencies
+#
+# Your model will be running remotely inside a container. We will be installing
+# all the model dependencies in the next step. We will also be "baking the model"
+# into the image using the script `download_stable_diffusion_models.py`.
+# This is technique that allows you to copy model files to
+# a worker more efficiently because they only need to be moved once.
+
+image = (
+    modal.Image.conda()
+    .apt_install(["curl"])
+    .run_commands(
+        [
+            "conda install xformers -c xformers/label/dev",
+            "conda install pytorch torchvision pytorch-cuda=11.7 -c pytorch -c nvidia",
+        ]
+    )
+    .run_commands(["pip install diffusers[torch] transformers ftfy accelerate"])
+    .run_commands(
+        [
+            "curl -L https://gist.github.com/luiscape/36a8cd29b8ed54cfbfcf56d51fe23cc0/raw/a6bf16996efe7c59114eea7944b0f99741d83d54/download_stable_diffusion_models.py | python"
```

Comment:
This seems pretty hacky – do you think you can use the `run_function` method on the image? It lets you run a Python function during image construction, which I think is what you need here?

Reply:
Yes, that's a good idea. It's better than piping that script.